### PR TITLE
SISRP-27162 - UX Dev: Student Overview - Finish Development on Transfer Credit

### PR DESCRIPTION
**Pull request by davidlimcheng**
https://jira.berkeley.edu/browse/SISRP-27162
Design at:  https://jira.berkeley.edu/browse/SISRP-16411

Finishing the Transfer Credit legacy left by @paulkerschen =)

* This feature really made me question `<table>`s and their design limitations.  For example, I had to do some whacky things to get a border in between the two rows on the front-end - I also had to use a three-column layout to preserve different spacing on different rows, when really a 2-column layout should have been fine.  
* With the addition of `MyAcademics::TransferCredit` for advisors only, this marks the first time `MyAcademics::FilteredForAdvisors` strays and becomes more than a subset of `MyAcademics::Merged`.  For the time being at least - students should be getting `TransferCredit` in a future release.

### CODE REVIEW

File: app/models/campus_solutions/transfer_credit.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  class TransferCredit < Proxy
+
+    include CampusSolutionsIdRequired
+
+    def xml_filename
+      'transfer_credit.xml'
+    end
+
+    def build_feed(response)
+      adjusted_units = (response.try(:[], 'ROOT').try(:[], "UC_TRANSFER_CREDITS").try(:[], "TRANSFER_CREDIT").try(:[], "UC_TRANSFER_CRSE_SCH").try(:[], "UNITS_ADJUST")).to_f
```
Comment (lyttam) [inline app/models/campus_solutions/transfer_credit.rb]: @davidlimcheng Do we have a preference here for single over double quotes?  I feel like I see single quotes used everywhere except where string interpolation is needed.  Not sure if it matters.

Comment (davidlimcheng) [inline app/models/campus_solutions/transfer_credit.rb]: @lyttam Yikes, I'm not sure what the preference is but I definitely shouldn't have mixed quote types in the same method.  I'll amend with single quotes.

**Comment by davidlimcheng** [comment]
@lyttam Amended, plus added endpoint to `verify_sis_endpoints.sh` script.

**Comment by redconfetti** [comment]
@davidlimcheng - Let's review this tomorrow morning

File: app/models/campus_solutions/transfer_credit.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  class TransferCredit < Proxy
+
+    include CampusSolutionsIdRequired
+
+    def xml_filename
+      'transfer_credit.xml'
+    end
+
+    def build_feed(response)
+      adjusted_units = (response.try(:[], 'ROOT').try(:[], 'UC_TRANSFER_CREDITS').try(:[], 'TRANSFER_CREDIT').try(:[], 'UC_TRANSFER_CRSE_SCH').try(:[], 'UNITS_ADJUST')).to_f
```
Comment (davidlimcheng) [inline app/models/campus_solutions/transfer_credit.rb]: If you're using the `if` and you know that the 'ROOT' is present, why not move these two variables, which are only used within the `if` code block... inside of that code block?

Comment (davidlimcheng) [inline app/models/campus_solutions/transfer_credit.rb]: Smeagol: Lousy ifeses! Should go in the blockses!!!!

File: app/models/my_academics/transfer_credit.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module MyAcademics
+  class TransferCredit
+    include AcademicsModule
+
+    def merge(data)
+      data[:transferCredit] = transfer_credit
+    end
+
+    def transfer_credit
+      response = CampusSolutions::TransferCredit.new(user_id: @uid).get
```
Comment (raydavis) [inline app/models/my_academics/transfer_credit.rb]: +1

**Comment by redconfetti** [comment]
@davidlimcheng - I was going to merge this, but I forgot that there are some modifications you can still make before we merge.

**Comment by davidlimcheng** [comment]
Made the changes we discussed, @redconfetti - should be good to go now.  Thanks for the good feedback.

**Pull request merged**